build.gradle

```diff
@@ -270,38 +270,97 @@ tasks.register('buildJars') {
+        events "passed", "skipped", "failed"
+    }
 
-    forkEvery = 1
-    enableAssertions = true
-    maxHeapSize = "1024m"
+    // Add summary reporting
+    afterSuite { testDescriptor, result ->
+        if (!testDescriptor.parent) {
+            println "Test Execution (${name}): ${result.resultType}" // Use task name
+            def summaryFields = ["${result.testCount} tests",
+                                 "${result.successfulTestCount} passed",
+                                 "${result.failedTestCount} failed",
+                                 "${result.skippedTestCount} skipped"]
+            println "Summary (${name}): " + summaryFields.join(", ") // Use task name
+        }
+    }
 
-    include "**/*Test.class"
+    include "**/*Test.class" // Include all potential tests first
     exclude "**/SplitInputStreamTest.class"
     exclude "**/JPF_*.class"
 
-    jvmArgs += ['--add-exports', 'java.base/jdk.internal.misc=ALL-UNNAMED', '--add-opens', 'java.base/jdk.internal.misc=ALL-UNNAMED', '--add-opens', 'java.base/java.lang=ALL-UNNAMED', '--add-opens', 'java.base/java.util=ALL-UNNAMED']
+    // Use JUnit 4 Categories to exclude singleThreadedTest
+    useJUnit {
+        excludeCategories 'gov.nasa.jpf.test.SingleThreadTest'
+    }
+}
+
+// Define Slow Test Task (renamed to singleThreadedTest)
+tasks.register('singleThreadedTest', Test) {
+    description = "Runs single threaded core regression tests (marked with @Category(singleThreadedTest.class)). These might not be parallelizable."
+    group = "JPF Test"
+
+    // Inherit configurations from the default test task
+    dependsOn buildJars
+    forkEvery = 1 // Explicitly set forkEvery
+    maxParallelForks = 1 // Run single threaded tests sequentially by default
+    enableAssertions = true // Explicitly set enableAssertions
+    maxHeapSize = "1024m" // Explicitly set maxHeapSize
+    jvmArgs = ['--add-exports', 'java.base/jdk.internal.misc=ALL-UNNAMED', '--add-opens', 'java.base/jdk.internal.misc=ALL-UNNAMED', '--add-opens', 'java.base/java.lang=ALL-UNNAMED', '--add-opens', 'java.base/java.util=ALL-UNNAMED'] // Explicitly set jvmArgs
 
     testLogging {
         events "passed", "skipped", "failed"
     }
 
+    // Add summary reporting
     afterSuite { testDescriptor, result ->
         if (!testDescriptor.parent) {
-            println "Test Execution: ${result.resultType}"
-
+            println "Test Execution (${name}): ${result.resultType}" // Use task name
             def summaryFields = ["${result.testCount} tests",
                                  "${result.successfulTestCount} passed",
                                  "${result.failedTestCount} failed",
                                  "${result.skippedTestCount} skipped"]
-
-            println "Summary: " + summaryFields.join(", ")
+            println "Summary (${name}): " + summaryFields.join(", ") // Use task name
         }
     }
+
+    include "**/*Test.class" // Include all potential tests first
```

Comment:
Limit this to URLClassLoaderTest and AssertTest for the single-threaded case, so the test method lookup of JUnit's test runner will be fast.